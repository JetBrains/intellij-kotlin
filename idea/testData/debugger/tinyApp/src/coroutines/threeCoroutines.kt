package threeCoroutines

import kotlin.random.Random

suspend fun main() {
    sequence {
        yield(239)
        sequence {
            //Breakpoint!
            yield(666)
        }.toList()
    }.toList()
}
