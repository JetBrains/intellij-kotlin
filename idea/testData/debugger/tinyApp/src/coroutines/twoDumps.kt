package twoDumps

suspend fun main() {
    foo()
    sequence<Int> {
        foo()
        yield(666)
    }.toList()
}

suspend fun foo() {
    val f = 239
    bar()
}

suspend fun bar() {
    var r = 1337
    //Breakpoint!
    r += 42
}

suspend fun SequenceScope<Int>.foo() {
    val k = 228
    bar()
}

suspend fun SequenceScope<Int>.bar() {
    var r = 1337
    //Breakpoint!
    r += 42
}