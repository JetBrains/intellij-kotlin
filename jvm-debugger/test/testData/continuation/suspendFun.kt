package coroutine1

import kotlin.coroutines.Continuation
import kotlin.coroutines.EmptyCoroutineContext
import kotlin.coroutines.startCoroutine

fun main() {
    val cnt = Continuation<Int>(EmptyCoroutineContext) { }
    val result = ::test1.startCoroutine(1, cnt)
    println(result)
}

suspend fun test1(i: Int): Int {
    val test1 = "a"
    a(test1)
    println(test1)
    return i
}

suspend fun a(aParam: String) {
    val a = "a"
    b(a)
    println(a)
}

suspend fun b(bParam: String) {
    val b = "b"
    // Breakpoint!
    println(b)
}