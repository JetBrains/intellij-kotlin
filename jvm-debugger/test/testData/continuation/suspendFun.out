LineBreakpoint created at suspendFun.kt:28
Run Java
Connected to the target VM
suspendFun.kt:28
Thread stack trace:
	b:28, SuspendFunKt (coroutine1)
		($completion, b, bParam)
	a:21, SuspendFunKt (coroutine1)
		($completion, $continuation, $result, a, aParam)
CoroutineNameIdState(name=coroutine, id=-1, state=UNKNOWN, dispatcher=null)
	test1:15, coroutine1.SuspendFunKt
		(i, test1)
	invoke:9, coroutine1.SuspendFunKt$main$result$1
		(continuation, p1, this)
	invokeSuspend:121, kotlin.coroutines.intrinsics.IntrinsicsKt__IntrinsicsJvmKt$createCoroutineUnintercepted$$inlined$createCoroutineFromSuspendFunction$IntrinsicsKt__IntrinsicsJvmKt$3
		($i$a$-createCoroutineFromSuspendFunction-IntrinsicsKt__IntrinsicsJvmKt$createCoroutineUnintercepted$2, it, result, this)
	resumeWith:33, kotlin.coroutines.jvm.internal.BaseContinuationImpl
		($i$a$-with-BaseContinuationImpl$resumeWith$1, $this$with, completion, current, param, result, this)
	startCoroutine:128, kotlin.coroutines.ContinuationKt
		($this$startCoroutine, completion, receiver)
	main:9, coroutine1.SuspendFunKt
		(cnt)
Disconnected from the target VM

Process finished with exit code 0
kotlin.Unit
